### PR TITLE
2325 testing needs to be hermetic against user config

### PR DESCRIPTION
**Pull request by lucas-wilkins**
Pytest uses config defaults now, I think. I'm not 100% sure this is actually doing anything though.

### CODE REVIEW

**Review by rozyczko — COMMENTED** [review]
I ran tests in this branch.
The default Exit SasView confirmation dialog still stops the testing.
Switching it off by clicking allows tests to continue.
This has nothing to do with user config but needs addressing.
Other changes to the user config seem to be saved/reloaded properly on test completion, assuming tests succeed without sasview seg faulting.

Code looks OK

**Comment by lucas-wilkins** [comment]
@rozyczko OK, will merge in your fix when you find it. Let me know if you can't.

**Comment by lucas-wilkins** [comment]
@rozyczko Merged your patch and turned on tests... waiting for a result

**Review by llimeht — APPROVED** [review]

File: src/sas/qtgui/MainWindow/UnitTesting/MainWindowTest.py [hunk 1]
```diff
@@ -99,12 +98,13 @@ def testExit(self, qapp, mocker):
...
+        mocker.patch.object(HidableDialog, 'exec', return_value=1)
 
         # Open, then close the main window
         screen_resolution = QtCore.QRect(0, 0, 640, 480)
         tmp_main = MainSasViewWindow(screen_resolution, None)
         tmp_main.close()
 
         # See that the MessageBox method got called
-        assert HidableDialog.exec.called
+        assert HidableDialog.called_once()
```
Comment (krzywon) [inline src/sas/qtgui/MainWindow/UnitTesting/MainWindowTest.py:110]: This is throwing an error locally (Win10). Should this be `HidableDialog.exec.called_once()`?

Error message:

> AttributeError: type object 'HidableDialog' has no attribute 'called_once'

**Comment by lucas-wilkins** [comment]
Probably, was part of Piotr's patch

File: src/sas/qtgui/MainWindow/UnitTesting/MainWindowTest.py (hunk 1 above)
Comment (rozyczko) [reply src/sas/qtgui/MainWindow/UnitTesting/MainWindowTest.py]: Yes, it should.

Merged the change.

**Comment by lucas-wilkins** [comment]
Thanks

**Pull request merged**